__tests__/app.test.js

```diff
@@ -43,109 +124,71 @@ describe("testing app GET /api/topics", () => {
-  test("should output 400 and a message stating the id is invalid", () => {
-    return request(app)
-      .get(`/api/articles/${badId}`)
-      .expect(400)
-      .then(({ body: { msg } }) => {
-        expect(msg).toBe("Your body or request has the wrong input type");
-      });
-  });
-  const badTypeBody = {
-    inc_votes: "NotType",
+describe("testing app GET /api/articles", () => {
+  const testArticleObj = {
+    article_id: expect.any(Number),
+    title: expect.any(String),
+    topic: expect.any(String),
+    author: expect.any(String),
+    body: expect.any(String),
+    created_at: expect.any(String),
+    votes: expect.any(Number),
+    comment_count: expect.any(Number),
   };
-  test("should output a 400 and a message stating the body is of wrong type", () => {
+  test("should return an array of article objects with correct properties, status 200 by default sorted by descending date", () => {
     return request(app)
-      .patch("/api/articles/1")
-      .send(badTypeBody)
-      .expect(400)
-      .then(({ body: { msg } }) => {
-        expect(msg).toBe("Your body or request has the wrong input type");
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const articles = body;
+        expect(articles).toHaveLength(12);
+        expect(articles[0].created_at).toEqual("2020-11-03T09:12:00.000Z");
+        expect(articles).toEqual(
+          expect.arrayContaining([expect.objectContaining(testArticleObj)])
+        );
       });
   });
 });
 
-describe("tests for malformed body", () => {
-  const emptyBody = {};
-  test("should output a 400 and a message stating the body is malformed/missing fields when given empty body", () => {
-    return request(app)
-      .patch("/api/articles/1")
-      .send(emptyBody)
-      .expect(400)
-      .then(({ body: { msg } }) => {
-        expect(msg).toBe("The body is missing fields or Malformed !");
-      });
-  });
-  const malformedBody = {
-    bad: 9999,
+describe("testing app /api/articles/:article_id/comments", () => {
+  const testCommentsObj = {
+    comment_id: expect.any(Number),
+    votes: expect.any(Number),
+    created_at: expect.any(String),
+    author: expect.any(String),
+    body: expect.any(String),
   };
-  test("should output a 400 and a message stating the body is malformed/missing fields when given malformed body", () => {
+  const articleId = 1;
+  test("should return an array of comments with correct properties", () => {
     return request(app)
-      .patch("/api/articles/1")
-      .send(malformedBody)
-      .expect(400)
-      .then(({ body: { msg } }) => {
-        expect(msg).toBe("The body is missing fields or Malformed !");
+      .get(`/api/articles/${articleId}/comments`)
+      .expect(200)
+      .then(({ body }) => {
+        const comments = body;
+        expect(comments).toHaveLength(11);
+        expect(comments).toEqual(
+          expect.arrayContaining([expect.objectContaining(testCommentsObj)])
```

Comment:
arrayContaining will stop checking at the first object if it finds an object that matches what where are expecting. It would also pass if this array contained elements not related to what we are expecting. A forEach would avoid these issues

Reply:
I just closed the pull request; I can change into a for Each